File: app-kotlin/src/main/java/fudi/freddy/mokitosample/mvp_sevice/AbstractPresent.java
package fudi.freddy.mokitosample.mvp_sevice;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */

public abstract class AbstractPresent<T extends MVPView> {

    T mvpView;

    public AbstractPresent(){

    }

    public void attach(T actView){
        mvpView = actView;
    }

    public void detach(){
        mvpView = null;
    }

    public T getView(){
        return mvpView;
    }

}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/mvp_sevice/AbstractPresent.kt
package fudi.freddy.mokitosample.mvp_sevice

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 7/25/20
 */
abstract class AbstractPresent<T : MVPView?> {
    var view: T? = null
    open fun attach(actView: T) {
        view = actView
    }

    open fun detach() {
        view = null
    }

}